doc/contributing/writing-tests.md

```diff
@@ -141,6 +141,26 @@ request. Interesting things to notice:
 
 ## General recommendations
 
+### Usage of `node:test`
+
+It is optional to use `node:test` in tests outside of testing the `node:test`
+module, as long as the functionality being tested is not a dependency of the
+`node:test` module. This ensures that a bug in the test runner doesn't impact
+the outcome of the underlying dependencies' test results.
+
+These dependencies are:
+
+- `node:async_hooks`
+- `node:child_process`
+- `node:fs`
+- ReadableStream in `node:streams`
```

Comment:
```suggestion
- `node:streams` with the exception of WebStreams
```

You can't really decouple `ReadableStream` from `WritableStream` as they are two sides of the same coin.